### PR TITLE
Add table aws_cloudfront_response_headers_policy. Closes #1098

### LINKED ISSUE #1098: Add table aws_cloudfront_response_headers_policy
**References**
Terraform resource - https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cloudfront_response_headers_policy

### PR DESCRIPTION
**Pull request by omerosaienni**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
omerosaienni@engineering ~/source-code/steampipe/steampipe-plugin-aws/aws-test(1098-add-table-aws_cloudfront_response_headers_policy)$ ./tint.js aws_cloudfront_response_headers_policy
No env file present for the current environment:  staging 
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined

SETUP: tests/aws_cloudfront_response_headers_policy []

PRETEST: tests/aws_cloudfront_response_headers_policy

TEST: tests/aws_cloudfront_response_headers_policy
Running terraform

Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # aws_cloudfront_response_headers_policy.named_test_resource will be created
  + resource "aws_cloudfront_response_headers_policy" "named_test_resource" {
      + comment = "test comment"
      + etag    = (known after apply)
      + id      = (known after apply)
      + name    = "turbottest42635"

      + cors_config {
          + access_control_allow_credentials = true
          + origin_override                  = true

          + access_control_allow_headers {
              + items = [
                  + "test",
                ]
            }

          + access_control_allow_methods {
              + items = [
                  + "GET",
                ]
            }

          + access_control_allow_origins {
              + items = [
                  + "test.example.comtest",
                ]
            }
        }

      + custom_headers_config {
          + items {
              + header   = "X-Permitted-Cross-Domain-Policies"
              + override = true
              + value    = "none"
            }
          + items {
              + header   = "X-Test"
              + override = true
              + value    = "none"
            }
        }

      + security_headers_config {

          + content_type_options {
              + override = true
            }

          + frame_options {
              + frame_option = "SAMEORIGIN"
              + override     = false
            }

          + referrer_policy {
              + override        = false
              + referrer_policy = "strict-origin-when-cross-origin"
            }

          + strict_transport_security {
              + access_control_max_age_sec = 31536000
              + override                   = false
            }

          + xss_protection {
              + mode_block = true
              + override   = false
              + protection = true
            }
        }
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Changes to Outputs:
  + account_id    = "11112222333"
  + etag          = (known after apply)
  + region_name   = "us-east-1"
  + resource_aka  = (known after apply)
  + resource_id   = (known after apply)
  + resource_name = "turbottest42635"
aws_cloudfront_response_headers_policy.named_test_resource: Creating...
aws_cloudfront_response_headers_policy.named_test_resource: Creation complete after 2s [id=3bdcfea9-4b72-4dcf-9fc8-a92ac979f286]

Apply complete! Resources: 1 added, 0 changed, 0 destroyed.

Outputs:

account_id = "11112222333"
etag = "E23ZP02F085DFQ"
region_name = "us-east-1"
resource_aka = "arn:aws:cloudfront::11112222333:response-headers-policy/3bdcfea9-4b72-4dcf-9fc8-a92ac979f286"
resource_id = "3bdcfea9-4b72-4dcf-9fc8-a92ac979f286"
resource_name = "turbottest42635"

Running SQL query: test-get-query.sql
[
  {
    "account_id": "11112222333",
    "arn": "arn:aws:cloudfront::11112222333:response-headers-policy/3bdcfea9-4b72-4dcf-9fc8-a92ac979f286",
    "e_tag": "E23ZP02F085DFQ",
    "id": "3bdcfea9-4b72-4dcf-9fc8-a92ac979f286",
    "name": "turbottest42635",
    "response_headers_policy_config": {
      "Comment": "test comment",
      "CorsConfig": {
        "AccessControlAllowCredentials": true,
        "AccessControlAllowHeaders": {
          "Items": [
            "test"
          ],
          "Quantity": 1
        },
        "AccessControlAllowMethods": {
          "Items": [
            "GET"
          ],
          "Quantity": 1
        },
        "AccessControlAllowOrigins": {
          "Items": [
            "test.example.comtest"
          ],
          "Quantity": 1
        },
        "AccessControlExposeHeaders": {
          "Items": null,
          "Quantity": 0
        },
        "AccessControlMaxAgeSec": null,
        "OriginOverride": true
      },
      "CustomHeadersConfig": {
        "Items": [
          {
            "Header": "X-Test",
            "Override": true,
            "Value": "none"
          },
          {
            "Header": "X-Permitted-Cross-Domain-Policies",
            "Override": true,
            "Value": "none"
          }
        ],
        "Quantity": 2
      },
      "Name": "turbottest42635",
      "SecurityHeadersConfig": {
        "ContentSecurityPolicy": {
          "ContentSecurityPolicy": null,
          "Override": null
        },
        "ContentTypeOptions": {
          "Override": true
        },
        "FrameOptions": {
          "FrameOption": "SAMEORIGIN",
          "Override": false
        },
        "ReferrerPolicy": {
          "Override": false,
          "ReferrerPolicy": "strict-origin-when-cross-origin"
        },
        "StrictTransportSecurity": {
          "AccessControlMaxAgeSec": 31536000,
          "IncludeSubdomains": false,
          "Override": false,
          "Preload": false
        },
        "XSSProtection": {
          "ModeBlock": true,
          "Override": false,
          "Protection": true,
          "ReportUri": null
        }
      }
    },
    "type": "custom"
  }
]
✔ PASSED

Running SQL query: test-list-query.sql
[
  {
    "account_id": "11112222333",
    "arn": "arn:aws:cloudfront::11112222333:response-headers-policy/3bdcfea9-4b72-4dcf-9fc8-a92ac979f286",
    "e_tag": "E23ZP02F085DFQ",
    "id": "3bdcfea9-4b72-4dcf-9fc8-a92ac979f286",
    "name": "turbottest42635",
    "response_headers_policy_config": {
      "Comment": "test comment",
      "CorsConfig": {
        "AccessControlAllowCredentials": true,
        "AccessControlAllowHeaders": {
          "Items": [
            "test"
          ],
          "Quantity": 1
        },
        "AccessControlAllowMethods": {
          "Items": [
            "GET"
          ],
          "Quantity": 1
        },
        "AccessControlAllowOrigins": {
          "Items": [
            "test.example.comtest"
          ],
          "Quantity": 1
        },
        "AccessControlExposeHeaders": {
          "Items": null,
          "Quantity": 0
        },
        "AccessControlMaxAgeSec": null,
        "OriginOverride": true
      },
      "CustomHeadersConfig": {
        "Items": [
          {
            "Header": "X-Test",
            "Override": true,
            "Value": "none"
          },
          {
            "Header": "X-Permitted-Cross-Domain-Policies",
            "Override": true,
            "Value": "none"
          }
        ],
        "Quantity": 2
      },
      "Name": "turbottest42635",
      "SecurityHeadersConfig": {
        "ContentSecurityPolicy": {
          "ContentSecurityPolicy": null,
          "Override": null
        },
        "ContentTypeOptions": {
          "Override": true
        },
        "FrameOptions": {
          "FrameOption": "SAMEORIGIN",
          "Override": false
        },
        "ReferrerPolicy": {
          "Override": false,
          "ReferrerPolicy": "strict-origin-when-cross-origin"
        },
        "StrictTransportSecurity": {
          "AccessControlMaxAgeSec": 31536000,
          "IncludeSubdomains": false,
          "Override": false,
          "Preload": false
        },
        "XSSProtection": {
          "ModeBlock": true,
          "Override": false,
          "Protection": true,
          "ReportUri": null
        }
      }
    },
    "type": "custom"
  }
]
✔ PASSED

Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:cloudfront::11112222333:response-headers-policy/3bdcfea9-4b72-4dcf-9fc8-a92ac979f286"
    ],
    "region": "us-east-1",
    "title": "turbottest42635"
  }
]
✔ PASSED

POSTTEST: tests/aws_cloudfront_response_headers_policy

TEARDOWN: tests/aws_cloudfront_response_headers_policy

SUMMARY:

1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  id,
  response_headers_policy_config ->> 'Comment' as description,
  type,
  last_modified_time
from
  aws_cloudfront_response_headers_policy;

+-------------------------------------------------------+--------------------------------------+-----------------------------------------------------------------------------------------------+---------+---------------------------+
| name                                                  | id                                   | description                                                                                   | type    | last_modified_time        |
+-------------------------------------------------------+--------------------------------------+-----------------------------------------------------------------------------------------------+---------+---------------------------+
| Managed-SimpleCORS                                    | 60669652-455b-4ae9-85a4-c4c02393f86c | Allows all origins for simple CORS requests                                                   | managed | 1970-01-01T12:00:00+12:00 |
| Managed-CORS-With-Preflight                           | 5cc3b908-e619-4b99-88e5-2cf7f45965bd | Allows all origins for CORS requests, including preflight requests                            | managed | 1970-01-01T12:00:00+12:00 |
| turbottest42635                                       | 3bdcfea9-4b72-4dcf-9fc8-a92ac979f286 | test comment                                                                                  | custom  | 2022-07-01T17:41:35+12:00 |
| test-policy-server-timing                             | 66479b2b-7cb9-4253-8371-5f68bed9845a |                                                                                               | custom  | 2022-06-29T22:10:36+12:00 |
| test-policy-security-headers-1                        | 2d2e8a4e-325a-4efd-be05-415283549599 |                                                                                               | custom  | 2022-06-29T22:07:24+12:00 |
| Managed-SecurityHeadersPolicy                         | 67f7725c-6f97-4210-82d7-5512b31e9d03 | Adds a set of security headers to every response                                              | managed | 1970-01-01T12:00:00+12:00 |
| test-policy-cors                                      | faf5c0a9-0fa5-43c2-ac79-142d52da785e | test policy with cors optional enabled                                                        | custom  | 2022-06-29T22:05:46+12:00 |
| Managed-CORS-and-SecurityHeadersPolicy                | e61eb60c-9c35-4d20-a928-2b84e02af89c | Allows all origins for simple CORS requests, and adds security headers                        | managed | 1970-01-01T12:00:00+12:00 |
| test-policy-security-headers-2                        | 83346797-0017-4a3a-8ef2-056bd052a224 | more security headers checking                                                                | custom  | 2022-06-29T22:09:41+12:00 |
| Managed-CORS-with-preflight-and-SecurityHeadersPolicy | eaab4381-ed33-4a86-88ca-d9558dc6cd63 | Allows all origins for CORS requests, including preflight requests, and adds security headers | managed | 1970-01-01T12:00:00+12:00 |
| test-policy                                           | f8cbe821-edc1-481b-a418-5de9651a163f | a test policy                                                                                 | custom  | 2022-06-29T22:04:15+12:00 |
+-------------------------------------------------------+--------------------------------------+-----------------------------------------------------------------------------------------------+---------+---------------------------+

> select
  name,
  id,
  response_headers_policy_config ->> 'Comment' as description,
  type,
  last_modified_time
from
  aws_cloudfront_response_headers_policy
where
  type = 'custom';

+--------------------------------+--------------------------------------+----------------------------------------+--------+---------------------------+
| name                           | id                                   | description                            | type   | last_modified_time        |
+--------------------------------+--------------------------------------+----------------------------------------+--------+---------------------------+
| test-policy-security-headers-2 | 83346797-0017-4a3a-8ef2-056bd052a224 | more security headers checking         | custom | 2022-06-29T22:09:41+12:00 |
| test-policy-cors               | faf5c0a9-0fa5-43c2-ac79-142d52da785e | test policy with cors optional enabled | custom | 2022-06-29T22:05:46+12:00 |
| test-policy-server-timing      | 66479b2b-7cb9-4253-8371-5f68bed9845a |                                        | custom | 2022-06-29T22:10:36+12:00 |
| test-policy                    | f8cbe821-edc1-481b-a418-5de9651a163f | a test policy                          | custom | 2022-06-29T22:04:15+12:00 |
| test-policy-security-headers-1 | 2d2e8a4e-325a-4efd-be05-415283549599 |                                        | custom | 2022-06-29T22:07:24+12:00 |
+--------------------------------+--------------------------------------+----------------------------------------+--------+---------------------------+

> select
  name,
  id,
  last_modified_time,
  now() as current_time
from
  aws_cloudfront_response_headers_policy
where
  last_modified_time >= (now() - interval '1' hour)
order by
  last_modified_time DESC;

+------------------------------------+--------------------------------------+---------------------------+---------------------------+
| name                               | id                                   | last_modified_time        | current_time              |
+------------------------------------+--------------------------------------+---------------------------+---------------------------+
| turbot-test-20200125-create-update | 5c1e654e-1245-48bb-ad8f-ba1af3740e48 | 2022-07-01T17:49:49+12:00 | 2022-07-01T17:54:40+12:00 |
+------------------------------------+--------------------------------------+---------------------------+---------------------------+
```
</details>

### CODE REVIEW

File: aws/table_aws_cloudfront_response_headers_policy.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+				Transform:   transform.FromField("ResponseHeadersPolicy.LastModifiedTime"),
+			},
+			{
+				Name:        "type",
+				Description: "The type of response headers policy, either managed (created by AWS) or custom (created in this AWS account).",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Type"),
+			},
+			{
+				Name:        "e_tag",
```
Comment (bigdatasourav) [inline aws/table_aws_cloudfront_response_headers_policy.go:62]:
```suggestion
				Name:        "etag",
```

File: aws/table_aws_cloudfront_response_headers_policy.go [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+	var id string
+
+	if h.Item != nil {
+		item := h.Item.(*cloudfront.ResponseHeadersPolicySummary)
+		id = *item.ResponseHeadersPolicy.Id
+	} else {
+		id = d.KeyColumnQuals["id"].GetStringValue()
+	}
+
+	if id == "" {
```
Comment (bigdatasourav) [inline aws/table_aws_cloudfront_response_headers_policy.go:161]: This is a separate hydrate call for a single column. We will not get `id` from key columns here. Please make the changes.

File: docs/tables/aws_cloudfront_response_headers_policy.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+They allow the user to define CORS, security, and custom response headers as a configuration setting in CloudFront.
+
+This table details the contents of the response header policies.
+
+**Important notes:**
+
+This table supports the optional quals `type`.
+Queries with optional quals are optimised to use additional filtering provided by the AWS API function.
+
+Omitting the field `e_tag` will improve the performace of the query.
```
Comment (bigdatasourav) [inline docs/tables/aws_cloudfront_response_headers_policy.md:15]:
```suggestion
```

File: docs/tables/aws_cloudfront_response_headers_policy.md [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+The reason behind this is due to `e_tag` requiring an addition API call to get the currrent `ETag` value.
```
Comment (bigdatasourav) [inline docs/tables/aws_cloudfront_response_headers_policy.md:16]:
```suggestion
```

File: docs/tables/aws_cloudfront_response_headers_policy.md (hunk 3 above)
Comment (omerosaienni) [reply docs/tables/aws_cloudfront_response_headers_policy.md:15]: Probably irrelevant as very few will use `etag`

I will remove comment entirely

**Review by bigdatasourav — APPROVED** [review]
LGTM

**Pull request merged**